src/core/storage/sqliteStorage/sqliteSession.ts

```diff
@@ -44,6 +47,20 @@ class SqliteSession {
 
   async open(storageName: string): Promise<void> {
     const connection = new SQLiteConnection(CapacitorSQLite);
+
+    const platform = Capacitor.getPlatform();
+    const isPlatformEncryption = platform !== "web";
+    const isEncryptInConfig = (await connection.isInConfigEncryption()).result;
+    const isEncryption = isPlatformEncryption && isEncryptInConfig;
+    if (isEncryption) {
+      const isSetSecret = (await connection.isSecretStored()).result;
+      if (!isSetSecret) {
+        const newBran = (await Agent.agent.getBranAndMnemonic()).bran;
+        await connection.setEncryptionSecret(newBran);
+        await SecureStorage.set(KeyStoreKeys.DB_ENCRYPTION_BRAN, newBran);
```

Comment:
I'd store this first, and then set the encryption secret on SQLite. That way we make sure whatever is used on the DB is always stored in the keystore in crash the app crashes in between calls (if it does, it'll replace the unused key store value with the new one on restart)

Reply:
done!